### ISSUE TITLE
warn about if conditions that will never/always be satisfied

### ISSUE THREAD

**Issue by dpiddy** [koalaman/shellcheck#457]
Given something like this, where `-z` was meant to be used with `$name`:

```bash
name="hello"
if [ -z "name" ]; then
  echo "your name is blank!"
fi
```

it would be great to get a warning that the supplied condition is always satisfied. Same for cases where the condition could never be satisfied, such as `[ -n "name" ]`.

cc @ricardochimal

**Comment by koalaman** [comment]
Definitely agree. There is already a check like this for `[ a = b ]` and it would be very useful to extend it.

**Comment by koalaman** [comment]
At least partially fixed in 7929a9d and 7e84ad03

**Comment by dpiddy** [comment]
Thanks!
